src/SingleLinkedList.h

```diff
@@ -307,22 +307,56 @@ int SingleLinkedList<T>::indexOf(T* element) {
 		return loc;
 }
 
+/*
+*Looks in the list to see if the element is there
+*returns true if found else false
+*/
 template <class T>
 bool SingleLinkedList<T>::contains(T* element) {
-	cout << "List contains Element (" << *element << ")." << endl;
+	 
+     if(isEmpty()) {
+	    cout << "List does not contains Element (" << *element << ")." << endl;
+        return false;
+    }
+	
+	Node<T>* currentNode = head; //Are we at the start
+	
+	while (currentNode != NULL) {
+		if (*(currentNode->data) == *element){
+		    cout << "List contains Element (" << *element << ")." << endl;
+			return true;
+	
+			
+		}
+		delete currentNode; 
```

Comment:
Remove this line. You'll delete the whole list and cause null pointer issues when we try and use them again.